### PR TITLE
Domains: display domain warnings for unverified domains

### PR DESCRIPTION
**Pull request by dzver**
Display warnings for domains with unverified emails. Not verifying a domain may lead to that domain being suspended, so it's a critical problem, just like expiration.

To test:
0. The easiest way to reproduce the case with domains that are not verified is by hacking `lib/domains/assembler.js` - set `isPendingIcannVerification` to `true` or `false`
1. On a site with 1 domain, go to `domains/manage/site` and check if the proper notice appears when `isPendingIcannVerification` is `true` and disappears when it's `false`.

screenshot 2016-03-23 19 08 34

(2) On a site with multiple domains, check if the notice displays a list of domains with unverified emails, each one with a clickable 'Fix now' link.

screenshot 2016-03-23 19 02 47

On the original PR it was suggested to me that I use two patterns to reduce code - both leading to InvariantViolation, so I didn't follow these suggestions.

This is the second half of #2916

### CODE REVIEW

File: client/my-sites/upgrades/components/domain-warnings/style.scss [hunk 1]
```diff
@@ -0,0 +1,12 @@
+.domain-warnings__unverified-domains {
+	a {
```
Comment (klimeryk) [inline client/my-sites/upgrades/components/domain-warnings/style.scss]: IMHO this style is unnecessary. We already have a style for inline links in notices and they should be bold to be a bit more distinctive. And the `text-decoration` rule causes a double underlining in Chrome:
screen shot 2016-03-26 at 19 26 02

There's an open issue about the [underline missing in Safari](https://github.com/Automattic/wp-calypso/issues/4321). Either way, you should rely on the global styles for the proper styling of `<a>` in notices.

**Comment by klimeryk** [comment]
Aside from the small visual bug mentioned above, I have two suggestions about the notices:

1. Don't make them dismissible - these are urgent, critical notices, they should be handled ASAP, no reason to allow the user to dismiss them. This also has the additional benefit of giving us more width for the copy ;)
2. Use a `NoticeAction` (see [sample code here](https://github.com/Automattic/wp-calypso/blob/master/client/components/notice/docs/example.jsx#L64)) instead of a smallish link. This makes the action that the user has to take more obvious, IMHO:
screen shot 2016-03-26 at 19 52 45
   For the multiple domains case, I'm afraid we are stuck with inline links, but that should be a much less frequent case.

Apart from the above, everything looks fine to me :)

**Comment by dzver** [comment]
Updated according to @klimeryk's proposal, which changes the look.

**Comment by klimeryk** [comment]
Thanks for removing the `a` style and using the `NoticeAction` component :bow: However, in order for it to be rendered as on the screenshot I posted in my previous comment, the text of the notice has to go to the `text` property of `Notice`, not as a child of it.
Just like [in the example](https://github.com/Automattic/wp-calypso/blob/master/client/components/notice/docs/example.jsx#L64):

```
<Notice
	status="is-warning"
	isCompact={ this.state.compactNotices ? true : null }
	showDismiss={ false }
	text={ "I'm an `is-warning` notice with an action." }>
		<NoticeAction href="#">
			{ "Update" }
		</NoticeAction>
</Notice>
```

And a `Notice` with a `NoticeAction` looks best if it has `showDismiss={ false }` :) As I've mentioned before, it's a very important notice and dismissing it here does not even make sense (and it will re-appear after refresh). And last but not least, other notices in this file have it disabled too :wink:

**Comment by klimeryk** [comment]
There were just two small issues left - in the "more-than-one-unverified-domain" scenario, the notice still had `showDismiss={ true }` and in the error for one domain there was an unnecessary full stop for the `NoticeAction` (`Fix now.`). I didn't want to bother you with those, so I've amended the commit myself :stuck_out_tongue: 
So, now, it's ready to :ship: (might need a squash, though)

**Pull request merged**